tests/integration_tests/functional/test_cpu_features.py

```diff
@@ -536,13 +536,37 @@ def _check_msr_values_are_equal(before_msrs_fname, after_msrs_fname, guest_kerne
     before = pd.read_csv(before_msrs_fname)
     after = pd.read_csv(after_msrs_fname)
 
+    # As a sanity check, make sure the numbers of implemented MSRs are the same
+    num_impl_before = (before["STATUS"] == "implemented").sum()
+    num_impl_after = (after["STATUS"] == "implemented").sum()
+    assert num_impl_before == num_impl_after
+
+    # Apply MSR exception list
     # pylint: disable=singleton-comparison
-    before_flt = before[before["MSR_ADDR"].isin(MSR_EXCEPTION_LIST) == False]
-    after_flt = after[after["MSR_ADDR"].isin(MSR_EXCEPTION_LIST) == False]
+    flt_before = before[before["MSR_ADDR"].isin(MSR_EXCEPTION_LIST) == False]
+    flt_after = after[after["MSR_ADDR"].isin(MSR_EXCEPTION_LIST) == False]
     # pylint: enable=singleton-comparison
-    diff = pd.concat([before_flt, after_flt]).drop_duplicates(keep=False)
 
-    assert diff.empty, f"\n{guest_kernel_name}:\n{diff}"
+    # Only consider implemented MSRs for comparison
+    impl_before = flt_before.loc[flt_before["STATUS"] == "implemented"].drop(
+        "STATUS", axis=1
+    )
+    impl_after = flt_after.loc[flt_after["STATUS"] == "implemented"].drop(
+        "STATUS", axis=1
+    )
+
+    # Find out if the values are the same
+    all_equal = (impl_before == impl_after).all(axis=None)
+
+    # Arrange the diff as a side by side comparison of values
+    not_equal = (impl_before != impl_after).any(axis=1)
+    impl_before.columns = ["MSR_ADDR", "Before"]
+    impl_after.columns = ["MSR_ADDR", "After"]
+    diff = pd.merge(
+        impl_before[not_equal], impl_after[not_equal], on=["MSR_ADDR", "MSR_ADDR"]
+    ).to_string()
+
+    assert all_equal, f"\n{guest_kernel_name}:\n{diff}"
```

Comment:
Same here, I would amend these improvements to the commit where you created this function.

Reply:
same as above